api/api.go

```diff
@@ -0,0 +1,213 @@
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/url"
+
+	"github.com/wtg/shuttletracker/log"
+
+	"fmt"
+	"github.com/gorilla/mux"
+	"github.com/wtg/shuttletracker/database"
+	"gopkg.in/cas.v1"
+	"gopkg.in/mgo.v2/bson"
+	"strings"
+)
+
+// Configuration holds the settings for connecting to outside resources.
+type Config struct {
+	GoogleMapAPIKey      string
+	GoogleMapMinDistance int
+	CasURL               string `env:"CAS_URL"`
+	Authenticate         bool   `env:"AUTHENTICATE" envDefault:"true"`
+	ListenURL            string
+}
+
+// App holds references to Mongo resources.
+type API struct {
+	cfg     Config
+	CasAUTH *cas.Client
+	CasMEM  *cas.MemoryStore
+	db      database.Database
+}
+
+// InitApp initializes the application given a config and connects to backends.
+// It also seeds any needed information to the database.
+func New(cfg Config, db database.Database) *API {
+	//Initialize cas connection
+	url, error := url.Parse(cfg.CasURL)
```

Comment:
To be consistent with the rest of your code you should renaming `error` to `err` here.